(#31)feat/get user list

gilyeon00 opened the pull request:
## 추가/수정한 기능 설명
- [x] reponse 순서가 바뀐 형태입니다.  이렇게 진행해도 될지, 혹은 더 좋은 방법이 있을 지 피드백 부탁드리겠습니다. 

<img width="712" alt="스크린샷 2023-04-04 오후 8 33 14" src="https://user-images.githubusercontent.com/52391627/229778942-7c1ce63b-b55a-4bb7-b975-6ba9c7381cbb.png">

<br>


## check list
- [x] issue number를 브랜치 앞에 추가 하였는가?
- [x] 추가/수정사항을 설명하였는가?

gilyeon00 committed:
[#31]Feat: 전체 User List return Service 구현

gilyeon00 committed:
[#31]Feat: 전체 User List Controller 구현
- 각각의 회원에 대한 HATEOAS 사용
- HATEOAS : REST API의 규약 중 "어플리케이션 상태에 대한 엔진으로써 하이퍼미디어"

Commit:
style: google-java-format

gilyeon00 committed:
[#31]Fix: fix return values to ResponseEntity and add status, selflink

Commit:
style: google-java-format

gilyeon00 committed:
Merge branch 'develop' of https://github.com/Team-Port/Voilio-Backend into (#31)feat/getUserList

kimich1218 commented on src/main/java/com/techeer/port/voilio/domain/user/dto/request/UserRequest.java:
```diff
@@ -1,3 +1,25 @@
+@Getter
+public class UserRequest {
+
+  private Long id;
+  private String email;
+  private String password;
+  private String nickname;
```
UserRuest에서 유효성 검사하는게 좋습니다! request parameter의 null 체크를 위해 @NotNull, @NotBlabk, @NotEmpty중 선택해서 추가해주세요!

kimich1218 reviewed (commented):
리뷰 남겼습니다!

gilyeon00 committed:
[#36]Feat: id를 통해 지정 User 가져오는 Service 로직 구현

gilyeon00 committed:
[#36]Feat: delete 되지 않은 사용자 select query문 통해 findUserById() 구현

gilyeon00 committed:
[#36]Feat: GET_USER_SUCCESS ResultCode 생성

gilyeon00 committed:
[#36]Feat: UserRequest에 user id 추가

gilyeon00 committed:
[#36]Feat: getUserById Controller 구현

gilyeon00 committed:
[#36]Fix: getUserList Controller 개별 link에 연결하는 controller 수정
- getUserList() to getUserById(id)

Commit:
style: google-java-format

gilyeon00 commented:
## 이후 추가된 사항
- [x] getUserById 추가 및 **지정 회원 출력 API 생성**
- [x] getUserById 추가에 따른 모든 회원 출력 API  개별 link에 
<img width="698" alt="스크린샷 2023-04-06 오전 1 24 10" src="https://user-images.githubusercontent.com/52391627/230143527-ffa8a65a-03f4-483f-9216-93ce578febf3.png">
연결된 Controller 수정 (getUserList -> getUserById)

kimich1218 commented on src/main/java/com/techeer/port/voilio/domain/user/service/UserService.java:
```diff
@@ -21,4 +24,13 @@ public void registerUser(UserRequest userRequest) throws Exception {
+    return new ArrayList<User>(userRepository.findAll());
+  }
+
+  public User getUserById(Long user_id) {
+    User user = userRepository.findUserById(user_id).orElseThrow(NotFoundUser::new);
```
변수명은 카멜케이스로 해야해요!

kimich1218 reviewed (commented):
리뷰 남겼습니다!

gilyeon00 committed:
Merge branch 'develop' of https://github.com/Team-Port/Voilio-Backend into (#31)feat/getUserList

gilyeon00 committed:
[#36]Fix: fix pull error
- edit getUserList response
- add userid into builder
- change field name to camelcase

Commit:
style: google-java-format
